rbac/common/base/base_message.py

```diff
@@ -18,18 +18,19 @@
 from rbac.common.crypto.keys import Key
 from rbac.common.crypto.keys import PUBLIC_KEY_PATTERN
 from rbac.common.sawtooth import batcher
-from rbac.common.sawtooth.state_client import StateClient
-from rbac.common.sawtooth.client_sync import ClientSync
-
+from rbac.common.sawtooth import client
+from rbac.common.sawtooth import state_client
+from rbac.common.base import base_processor as processor
 
 LOGGER = logging.getLogger(__name__)
 
 
+# pylint: disable=too-many-public-methods
 class BaseMessage:
     def __init__(self):
-        """Objects and methods shared across message libraries"""
-        self.client = ClientSync()
-        self.state = StateClient()
+        """Base class for all messages"""
+        self._message_type_name = batcher.get_message_type_name(self.message_type)
+        processor.register_message_handler(self)
```

Comment:
The transaction processor executes the messages; thus it needs to know what they are.

Reply:
Previously the transaction processor discovered the message handlers. However, you explicitly disliked that pattern, thus the above is an alternative solution.

Reply:
Understood. This is an improvement over that. I am just not sure about having a base class register with another module and then be extended and ultimately instantiated. It confuses me because I understand a base class as nothing more than common features to be customized as needed in subclassing for operations on the interface (polymorphism).

I could be missing something here. We should discuss.

Reply:
The base class isn't registering, the parent classes are.